modules/configuration/src/main/java/org/apache/ignite/configuration/SystemConfiguration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ignite.configuration.internal.DynamicConfiguration;
+import org.apache.ignite.configuration.presentation.json.JsonPresentation;
+
+/** */
+public class SystemConfiguration {
+    /** */
+    private final Map<String, Configurator<?>> configs = new HashMap<>();
+
+    /** */
+    private final Map<String, ConfigurationTree<?, ?>> multirootConfiguration = new HashMap<>();
+
+    /** */
+    private final ConfigurationPresentation<String> presentation = new JsonPresentation(configs);
+
+    /** */
+    public <T extends DynamicConfiguration<?, ?, ?>> void registerConfigurator(Configurator<T> unitConfig) {
+        String key = unitConfig.getRoot().key();
+
+        configs.put(key, unitConfig);
+
+        multirootConfiguration.put(key, unitConfig.getRoot());
+    }
+
+    /** */
+    public <V, C, T extends ConfigurationTree<V, C>> T getConfiguration(RootKey<T> rootKey) {
```

Comment:
If we are going to implement a configuration class per root then this method is not longer needed

Reply:
We may stick with this approach until more complex modules are developed and see how it works against "configuration per root" approach. Then pick the best from API perspective.